Rotateauth for databases

Bonusree opened the pull request.

ArnobKumarSaha commented on docs/guides/postgres/rotate-authentication/rotateauth.md:
```diff
@@ -0,0 +1,434 @@
+  docs_{{ .version }}:
+    identifier: pg-rotate-auth-details
+    name: Rotate Authentication Postgres
+    parent: pg-restart
```
parent "pg-restart" ?

ArnobKumarSaha commented on docs/guides/postgres/rotate-authentication/rotateauth.md:
````diff
@@ -0,0 +1,434 @@
+
+  Let's create the `PostgresOpsRequest` CR we have shown above,
+  ```shell
+    $ kubectl apply -f https://github.com/kubedb/docs/raw/v2025.5.30/docs/examples/postgres/rotate-auth/postgres-rotate-auth-generated.yaml
````
The version (v2025.5.30) should be templatized.

ArnobKumarSaha commented on docs/guides/postgres/rotate-authentication/rotateauth.md:
````diff
@@ -0,0 +1,434 @@
+   $  kubectl get secret -n demo quick-postgres-auth -o jsonpath='{.data.password}' | base64 -d
+   zGB9GF!NXwI.2HP9⏎                       
+  ```
+Also, there will be two more new keys in the secret that stores the previous credentials. The keys are username.prev and password.prev. You can find the secret and its data by running the following command:
````
Write `username.prev` & `password.prev`; instead of username.prev & password.prev.

Simlilar fixes for all other cases.

ArnobKumarSaha commented on docs/guides/mongodb/rotate-auth/rotateauth.md:
````diff
@@ -0,0 +1,493 @@
+Let's create the `MongoDBOpsRequest` CR we have shown above,
+
+```shell
+$ kubectl apply -f https://github.com/kubedb/docs/raw/v2025.6.30/docs/examples/mongodb/rotate-auth/rotate-auth-user.yaml
````
By templeting, I meant somehting like this : 
`v2025.6.30` ->  `{{ .version }}`

Bonusree committed:
RotateAuth for databases

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
templetize done

Signed-off-by: Bonusree <bonusree@appscode.com>

ArnobKumarSaha commented:
Lets add rotateAuth docs for all these tick marked databases
https://github.com/kubedb/internal/issues/1193

Bonusree committed:
rotateauth for mariadb

Signed-off-by: Bonusree <bonusree@appscode.com>

ArnobKumarSaha committed:
Update petset tag in kubedb-upgrade process

Signed-off-by: Arnob kumar saha <arnob@appscode.com>

Bonusree committed:
some changes in mariadb

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
corrected ci problems

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
rotateauth for druid

Bonusree committed:
rotateauth for mssqlserver added

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
rotateauth for elasticsearch is added

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
rotateauth for pgbouncer

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
rotateauth for pgpool

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
redis rotate auth

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
rotateauth for solr

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
cli issues fixed

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
rotateauth for perconaxtradb

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
cli error resolved

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
remove next step from percona

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
failover started for mysql

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
here is failover for mysql

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
rename index file

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
Merge branch 'failover' into rotateauth

Bonusree committed:
fixed rotate auth

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
solr fixed

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
mariadb review resolve

Signed-off-by: Bonusree <bonusree@appscode.com>

refat75 commented on docs/guides/mariadb/rotate-auth/overview/index.md:
```diff
@@ -0,0 +1,10 @@
+---
```
I suppose this `index.md` is for overview section. As overview is not added, so you can remove this part. This creates an extra section without any content.

refat75 commented on docs/guides/mariadb/rotate-auth/rotateauth.md:
```diff
@@ -0,0 +1,433 @@
+**Rotate Authentication** is a feature of the KubeDB Ops-Manager that allows you to rotate a `MariaDB` user's authentication credentials using a `MariaDBOpsRequest`. There are two ways to perform this rotation.
+
+1. **Operator Generated:** The KubeDB operator automatically generates a random credential, updates the existing secret with the new credential, and does not provide the secret details directly to the user.
+2. **User Defined:** The user can create their own credentials by defining a Secret of type `kubernetes.io/basic-auth` containing the desired `username` and `password`, and then reference this Secret in the `MariaDBOpsRequest`.
```
In MariaDB, it is forbidden to change the `username`. Only `password` can be changed. Remove the `username` part. Also mention that `username` can not be changed.

refat75 commented on docs/guides/mariadb/rotate-auth/rotateauth.md:
```diff
@@ -0,0 +1,433 @@
+  version: "10.5.23"
+  storageType: Durable
+  storage:
+    storageClassName: "standard"
```
Remove `storageClassName`

Bonusree replied:
it's already given in 'quickstart' and I use refer from this . I think I should not delete this portion

Bonusree committed:
resolved review for mariadb

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
weight, rename fixed for all dbs

Signed-off-by: Bonusree <bonusree@appscode.com>

fr-sarker commented on docs/guides/druid/rotate-auth/guide.md:
```diff
@@ -0,0 +1,606 @@
+Here,
+
+- `spec.databaseRef.name` specifies that we are performing rotate authentication operation on `druid-quickstart`cluster.
+- `spec.type` specifies that we are performing `RotateAuth` on postgres.
```
Here Change the name postgres to druid

fr-sarker commented on docs/guides/elasticsearch/rotateauth/rotateauth.md:
```diff
@@ -0,0 +1,637 @@
+Here,
+
+- `spec.databaseRef.name` specifies that we are performing rotate authentication operation on `sample-es`cluster.
+- `spec.type` specifies that we are performing `RotateAuth` on postgres.
```
Here change the name postgres to elasticsearch

Bonusree committed:
resolve error druid, elastic, mssql, redis

Signed-off-by: Bonusree <bonusree@appscode.com>

HiranmoyChowdhury commented on docs/guides/pgbouncer/rotateauth/rotateauth.md:
```diff
@@ -0,0 +1,469 @@
+
+> We have designed this tutorial to demonstrate a production setup of KubeDB managed PgBouncer. If you just want to try out KubeDB, you can bypass some of the safety features following the tips [here](/docs/guides/pgbouncer/quickstart/quickstart.md#tips-for-testing).
+
+## Create a PgBouncer database
```
add a section "prepare postgres" before creating pgbouncer

HiranmoyChowdhury commented on docs/guides/pgpool/configuration/using-podtemplate.md:
```diff
@@ -507,7 +507,7 @@ Containers:
     Mounts:
       /config from pgpool-config (rw)
-      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-69qx2 (ro)
+      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-69qx2 (rotateauth)
```
why?

HiranmoyChowdhury commented on docs/guides/pgbouncer/rotateauth/rotateauth.md:
```diff
@@ -0,0 +1,469 @@
+
+#### 1. Using operator generated credentials:
+
+In order to rotate authentication to the PgBouncer using operator generated, we have to create a `PgBouncerOpsRequest` CRO with `RotateAuth` type. Below is the YAML of the `PgBouncerOpsRequest` CRO that we are going to create,
```
is there any specific meaning of CRO. Saw you used this in pgbouncer and pgpool multiple time

HiranmoyChowdhury commented on docs/guides/pgpool/rotateauth/rotateauth.md:
```diff
@@ -0,0 +1,228 @@
+  Timeout:  5m
+  Type:     RotateAuth
+Status:
+  Phase:  Pending
```
why is this pending after opsRequest being successful?

HiranmoyChowdhury commented on docs/guides/pgpool/rotateauth/rotateauth.md:
````diff
@@ -0,0 +1,228 @@
+At first, we need to create a secret with kubernetes.io/basic-auth type using custom username and password. Below is the command to create a secret with kubernetes.io/basic-auth type,
+
+```shell
+$  kubectl create secret generic quick-pb-user-auth -n pool \
````
change the name pb to pp

HiranmoyChowdhury commented on docs/guides/pgpool/rotateauth/rotateauth.md:
```diff
@@ -0,0 +1,228 @@
+  Timeout:  5m
+  Type:     RotateAuth
+Status:
+  Phase:  Pending
```
how this is in pending state?

Bonusree committed:
resolved review of pgpool&pgbouncer, make changes in name

Signed-off-by: Bonusree <bonusree@appscode.com>

1gtm committed:
Prepare for release v2025.7.30-rc.0 (#796)

ProductLine: KubeDB
Release: v2025.7.30-rc.0
Release-tracker: https://github.com/kubedb/CHANGELOG/pull/115
Signed-off-by: 1gtm <1gtm@appscode.com>

Neaj-Morshad-101 commented on docs/examples/mssqlserver/rotate-auth/MSSQLServer-rotate-auth-generated.yaml:
```diff
@@ -0,0 +1,11 @@
+apiVersion: ops.kubedb.com/v1alpha1
```
update the file name

Neaj-Morshad-101 commented on docs/guides/mssqlserver/rotate-auth/_index.md:
```diff
@@ -0,0 +1,10 @@
+menu:
+  docs_{{ .version }}:
+    identifier: ms-rotate-auth
+    name: Rotate Authentication MSSQLServer
```
remove `MSSQLServer`

Neaj-Morshad-101 commented on docs/guides/mssqlserver/rotate-auth/rotateauth.md:
```diff
@@ -0,0 +1,485 @@
+
+**Rotate Authentication** is a feature of the KubeDB Ops-Manager that allows you to rotate a `MSSQLServer` user's authentication credentials using a `MSSQLServerOpsRequest`. There are two ways to perform this rotation.
+
+1. **Operator Generated:** The KubeDB operator automatically generates a random credential, updates the existing secret with the new credential, and does not provide the secret details directly to the user.
```
does not provide the secret details directly to the user?

Neaj-Morshad-101 commented on docs/guides/mssqlserver/rotate-auth/rotateauth.md:
```diff
@@ -0,0 +1,485 @@
+
+1. **Operator Generated:** The KubeDB operator automatically generates a random credential, updates the existing secret with the new credential, and does not provide the secret details directly to the user.
+2. **User Defined:** The user can create their own credentials by defining a Secret of type 
+`kubernetes.io/basic-auth` containing the desired `password`, and then reference this Secret in the `MSSQLServerOpsRequest`.
```
in the `MSSQLServerOpsRequest` CR.

Neaj-Morshad-101 commented on docs/guides/mssqlserver/rotate-auth/rotateauth.md:
````diff
@@ -0,0 +1,485 @@
+namespace/demo created
+```
+
+> **Note:** YAML files used in this tutorial are stored in [docs/examples/mssqlserver](/docs/examples/mssqlserver/update-version) directory of [kubedb/docs](https://github.com/kube/docs) repository.
````
update-version -> rotate-auth

Neaj-Morshad-101 commented on docs/guides/mssqlserver/rotate-auth/rotateauth.md:
````diff
@@ -0,0 +1,485 @@
+$ kubectl create -f https://github.com/kubedb/docs/raw/{{< param "info.version" >}}/docs/examples/mssqlserver/quickstart/mssqlserver-quickstart.yaml
+mssqlserver.kubedb.com/mssqlserver-quickstart created
+```
+Now, wait until quick-MongoDB has status Ready. i.e,
````
MongoDB!

Neaj-Morshad-101 commented on docs/guides/mssqlserver/rotate-auth/rotateauth.md:
````diff
@@ -0,0 +1,485 @@
+mssqlserver-quickstart   2022-cu12   Ready    75m
+```
+## Verify authentication
+The user can verify whether they are authorized by executing a query directly in the database. To do this, the user needs `username` and `password` in order to connect to the database using the `kubectl exec` command. Below is an example showing how to retrieve the credentials from the Secret.
````
in order to connect to the database ~~using the `kubectl exec` command.~~

Neaj-Morshad-101 commented on docs/guides/mssqlserver/rotate-auth/rotateauth.md:
```diff
@@ -0,0 +1,485 @@
+"mssqlserver-quickstart-auth"
+$ kubectl get secret -n demo mssqlserver-quickstart-auth -o jsonpath='{.data.username}' | base64 -d
+sa⏎  
+$ kubectl get secret -n demo mssqlserver-quickstart-auth -o jsonpath='{.data.password}' | base64 -d
```
We can verify the change by connecting using both the old and new credentials.

Neaj-Morshad-101 commented on docs/guides/mssqlserver/rotate-auth/rotateauth.md:
```diff
@@ -0,0 +1,485 @@
+
+- `spec.databaseRef.name` specifies that we are performing rotate authentication operation on `mssqlserver-quickstart`cluster.
+- `spec.type` specifies that we are performing `RotateAuth` on MSSQLServer.
+- `spec.authentication.secretRef.name` specifies that we are using `quick-mssql-user-auth` as `spec.authSecret.name` for authentication.
```
we are using `quick-mssql-user-auth` as `spec.authSecret.name` for authentication. -> 
we want to use  `quick-mssql-user-auth` for database authentication.

Neaj-Morshad-101 commented on docs/guides/mssqlserver/rotate-auth/rotateauth.md:
````diff
@@ -0,0 +1,485 @@
+$ kubectl get secret -n demo quick-mssqlserver-user-auth -o go-template='{{ index .data "password.prev" }}' | base64 -d
+zTBVvzgoEb2qUe3X⏎ 
+```
+
````
verify the change by connecting using both the old and new credentials.

1gtm committed:
Prepare for release v2025.7.31 (#799)

ProductLine: KubeDB
Release: v2025.7.31
Release-tracker: https://github.com/kubedb/CHANGELOG/pull/116
Signed-off-by: 1gtm <1gtm@appscode.com>

HiranmoyChowdhury committed:
Valkey and External Client Connection (#795)

Signed-off-by: HiranmoyChowdhury <hiranmoy@appscode.com>

Bonusree committed:
correct the definition

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
resolve mssqlserver

Signed-off-by: Bonusree <bonusree@appscode.com>

ArnobKumarSaha commented on docs/guides/singlestore/configuration/podtemplating/index.md:
```diff
@@ -600,7 +600,7 @@ Containers:
       /scripts from init-scripts (rw)
       /var/lib/memsql from data (rw)
-      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-htm2z (ro)
+      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-htm2z (rotateauth)
```
Fix this typo

sabbir-hossain70 committed:
Add Cassandra Doc (#789)

Signed-off-by: sabbir <sabbir@appscode.com>

Neaj-Morshad-101 committed:
Add MSSQL Init Database Support via Scripts doc (#797)

Signed-off-by: Neaj Morshad <neaj@appscode.com>

sheikh-arman committed:
Distributed MariaDB (#798)

Signed-off-by: SK Ali Arman <arman@appscode.com>

refat75 committed:
Add maxscale volume expansion doc (#790)

Signed-off-by: Rabbani Islam Refat <refat@appscode.com>

Bonusree committed:
ro->rotateauth

Bonusree committed:
start ignite

Signed-off-by: Bonusree <bonusreedatta820@gmail.com>

1gtm committed:
Use Go 1.25 (#800)

Signed-off-by: 1gtm <1gtm@appscode.com>

Bonusree committed:
rabbitmq started

Signed-off-by: Bonusree <bonusreedatta820@gmail.com>

Bonusree committed:
rabbitmq rotateauth is ready for review

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
mysql rotateauth

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
try to fix build and mysql

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
resolve link error

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
resolve url error

Signed-off-by: Bonusree <bonusree@appscode.com>

refat75 commented on docs/guides/mysql/rotate-auth/_index.md:
```diff
@@ -1,495 +1,10 @@
-weight: 10
+  docs_{{ .version }}:
+    identifier: guides-mysql-rotate-auth
+    name: Rotate Authentication MySQL
```
Rotate Authentication MySQL -> Rotate Authentication

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
```diff
@@ -2,19 +2,22 @@
 # Rotate Authentication of Mysql
 
-**Rotate Authentication** is a feature of the KubeDB Ops-Manager that allows you to rotate a `Mysql` user's authentication credentials using a `MySQLopsrequest`. There are two ways to perform this rotation.
+**Rotate Authentication** is a feature of the KubeDB Ops-Manager that allows you to rotate a `Mysql`
```
Mysql --> MySQL

Change all the similar occurence.

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
```diff
@@ -2,19 +2,22 @@
 section_menu_id: guides
 ---
 
 # Rotate Authentication of Mysql
```
Mysql --> MySQL

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
```diff
@@ -2,19 +2,22 @@
 
-**Rotate Authentication** is a feature of the KubeDB Ops-Manager that allows you to rotate a `Mysql` user's authentication credentials using a `MySQLopsrequest`. There are two ways to perform this rotation.
+**Rotate Authentication** is a feature of the KubeDB Ops-Manager that allows you to rotate a `Mysql`
+user's authentication credentials using a `MySQLopsrequest`. There are two ways to perform this
```
MySQLopsrequest --> MySQLOpsRequest

Change all the similar occurence.

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
````diff
@@ -0,0 +1,595 @@
+8.4.3           8.4.3     Official       ghcr.io/appscode-images/mysql:8.4.3-oracle                 25h
+9.0.1           9.0.1     Official       ghcr.io/appscode-images/mysql:9.0.1-oracle                 25h
+9.1.0           9.1.0     Official       ghcr.io/appscode-images/mysql:9.1.0-oracle                 25h
+```
````
The above list contain deprecated images. Which is unnecessary.  Replace with the following content.

```bash
$ kubectl get mysqlversion
NAME            VERSION   DISTRIBUTION   DB_IMAGE                                      DEPRECATED   AGE
5.7.42-debian   5.7.42    Official       ghcr.io/appscode-images/mysql:5.7.42-debian                12d
5.7.44          5.7.44    Official       ghcr.io/appscode-images/mysql:5.7.44-oracle                12d
8.0.31-innodb   8.0.31    MySQL          ghcr.io/appscode-images/mysql:8.0.31-oracle                12d
8.0.35          8.0.35    Official       ghcr.io/appscode-images/mysql:8.0.35-oracle                12d
8.0.36          8.0.36    Official       ghcr.io/appscode-images/mysql:8.0.36-debian                12d
8.1.0           8.1.0     Official       ghcr.io/appscode-images/mysql:8.1.0-oracle                 12d
8.2.0           8.2.0     Official       ghcr.io/appscode-images/mysql:8.2.0-oracle                 12d
8.4.2           8.4.2     Official       ghcr.io/appscode-images/mysql:8.4.2-oracle                 12d
8.4.3           8.4.3     Official       ghcr.io/appscode-images/mysql:8.4.3-oracle                 12d
9.0.1           9.0.1     Official       ghcr.io/appscode-images/mysql:9.0.1-oracle                 12d
9.1.0           9.1.0     Official       ghcr.io/appscode-images/mysql:9.1.0-oracle                 12d

```

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
````diff
@@ -0,0 +1,595 @@
+9.1.0           9.1.0     Official       ghcr.io/appscode-images/mysql:9.1.0-oracle                 25h
+```
+
+## Create a Mysql server
````
server -> Database

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
```diff
@@ -0,0 +1,595 @@
+
+## Create a Mysql server
+
+KubeDB implements a `Mysql` CRD to define the specification of a Mysql server. Below is the `Mysql`
```
server -> database

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
````diff
@@ -0,0 +1,595 @@
+mysql.kubedb.com/mysql-quickstart created
+```
+
+## Verify authentication
````
authentication --> Authentication

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
```diff
@@ -0,0 +1,595 @@
+If you can access the data table and run queries, it means the secrets are working correctly.
+## Create RotateAuth MySQLopsrequest
+
+#### 1. Using operator generated credentials:
```
Write title using title case.

1. Using operator generated credentials: --> 1. Using Operator Generated Credentials:

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
````diff
@@ -0,0 +1,595 @@
+```
+Here,
+
+- `spec.databaseRef.name` specifies that we are performing rotate authentication operation on `mysql-quickstart` cluster.
````
Here we deployed mysql in standalone mode it is not cluster. 
cluster --> instance

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
````diff
@@ -0,0 +1,595 @@
+
+```
+The above output shows that the password has been changed successfully. The previous username & password is stored for rollback purpose.
+#### 2. Using user created credentials
````
Use title case.
#### 2. Using User Created Credentials

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
````diff
@@ -0,0 +1,595 @@
+```
+Here,
+
+- `spec.databaseRef.name` specifies that we are performing rotate authentication operation on `mysql-quickstart`cluster.
````
cluster --> instance.

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
```diff
@@ -0,0 +1,595 @@
+## Cleaning up
+
+To clean up the Kubernetes resources you can delete the CRD or namespace.
+Or, you can delete one by one resource by their name by this tutorial, run:
```
"Or, you can delete one by one resource by their name by this tutorial"  -->
Alternatively, you can delete individual resources by name. To do so, run:

Bonusree committed:
resolve review of mysql

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
memcached rotateauth added

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
delete the wrong url

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
mysql url error fixed

Signed-off-by: Bonusree <bonusree@appscode.com>

Evanraisul commented on docs/guides/memcached/rotate-auth/rotateauth.md:
```diff
@@ -0,0 +1,523 @@
+
+- Now, install KubeDB cli on your workstation and KubeDB operator in your cluster following the steps [here](/docs/setup/README.md). Make sure install with helm command including `--set global.featureGates.Memcached=true` to ensure Memcached CRD installation.
+
+- To configure TLS/SSL in `Memcached`, `KubeDB` uses `cert-manager` to issue certificates. So first you have to make sure that the cluster has `cert-manager` installed. To install `cert-manager` in your cluster following steps [here](https://cert-manager.io/docs/installation/kubernetes/).
```
Why need tls information here?

Evanraisul commented on docs/guides/memcached/rotate-auth/rotateauth.md:
````diff
@@ -0,0 +1,523 @@
+quit
+Connection closed by foreign host.
+```
+Your credentials have been stored successfully, so everything’s working.
````
stored -> updated/rotated

Evanraisul commented on docs/guides/memcached/rotate-auth/rotateauth.md:
````diff
@@ -0,0 +1,523 @@
+```shell
+$ kubectl get Memcachedopsrequest -n demo
+NAME                          TYPE         STATUS       AGE
+mcops-rotate-auth-generated   RotateAuth   Successful   19h
````
Remove the old one from here.

Evanraisul commented on docs/guides/memcached/rotate-auth/rotateauth.md:
````diff
@@ -0,0 +1,523 @@
+quit
+Connection closed by foreign host.
+```
+Your credentials have been stored successfully, so everything’s working.
````
stored -> updated/rotated

Evanraisul commented on docs/guides/memcached/rotate-auth/rotateauth.md:
````diff
@@ -0,0 +1,523 @@
+                  
+```
+
+The above output shows that the password has been changed successfully. The previous username & password is stored in the secret for rollback purpose.
````
changed -> updated/rotated

Bonusree committed:
change

Signed-off-by: Bonusree <bonusree@appscode.com>

Evanraisul reviewed (changes requested):
Where is the "overview.md" file in rotateAuth docs?
Can check our others OpsRequest for better understanding. Every opsRequest folder has "overview.md" file.

Bonusree committed:
Merge branch 'mariadb-failover' into rotateauth

Bonusree committed:
resolve review memcached rotateauth

Signed-off-by: Bonusree <bonusree@appscode.com>

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
```diff
@@ -0,0 +1,542 @@
+weight: 10
+menu_name: docs_{{ .version }}
+section_menu_id: guides
+---
```
Replace the above section with following content. There is some issues with indentation .
```
title: MySQL Rotateauth Guide
menu:
  docs_{{ .version }}:
    identifier: mysql-rotate-auth-guide
    name: Guide
    parent: guides-mysql-rotate-auth
    weight: 10
menu_name: docs_{{ .version }}
section_menu_id: guides
```

Bonusree committed:
try to fix visually

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
intendation

Signed-off-by: Bonusree <bonusree@appscode.com>

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
````diff
@@ -0,0 +1,542 @@
+$ kubectl create -f https://github.com/kubedb/docs/raw/{{< param "info.version" >}}/docs/guides/mysql/quickstart/yamls/quickstart-v1.yaml
+mysql.kubedb.com/mysql-quickstart created
+```
+
````
Add the following section.

Now, wait until mysql-quickstart has status Ready. i.e,
``` bash
$ kubectl get mysql -n demo -w
NAME             VERSION   STATUS   AGE
mysql-quickstart 10.5.23   Ready    30m
```

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
````diff
@@ -0,0 +1,542 @@
+mysql.kubedb.com/mysql-quickstart created
+```
+
+##Create a Mysql Database
````
Replace this line with `## Verify Authentication`

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
````diff
@@ -0,0 +1,542 @@
+```
+
+If you can access the data table and run queries, it means the secrets are working correctly.
+## Create RotateAuth MySQLopsrequest
````
MySQLopsrequest --> MySQLOpsRequest

refat75 commented on docs/guides/mysql/rotate-auth/guide.md:
```diff
@@ -0,0 +1,542 @@
+#### 1. Using Operator Generated Credentials:
+
+In order to rotate authentication to the MySQL using operator generated, we have to create a 
+`MySQLopsrequest` CR with `RotateAuth` type. Below is the YAML of the `MySQLopsrequest` CRO that
```
MySQLopsrequest --> MySQLOpsRequest

Change every occurence.

Bonusree committed:
resolve mysql review 2

Signed-off-by: Bonusree <bonusree@appscode.com>

ShuvoKumarMondal commented on docs/guides/solr/rotateauth/rotateauth.md:
```diff
@@ -0,0 +1,372 @@
+
+## Create RotateAuth SolrOpsRequest
+
+#### 1. Using operator generated credentials:
```
Create db before applying ops-req if not created. (also add overview.md like other ops-request, follow other db)

ShuvoKumarMondal commented on docs/guides/solr/rotateauth/rotateauth.md:
````diff
@@ -0,0 +1,372 @@
+To clean up the Kubernetes resources you can delete the CRD or namespace.
+Or, you can delete one by one resource by their name by this tutorial, run:
+
+```shell
````
clean db?

Bonusree committed:
review & overview added in solr

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
overview added

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
solr db clean resolve

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
ignite started

Signed-off-by: Bonusree <bonusree@appscode.com>

Bonusree committed:
Merge remote-tracking branch 'origin/master' into rotateauth

Bonusree committed:
rotateauth for clickhouse

Signed-off-by: Bonusree <bonusree@appscode.com>

ArnobKumarSaha commented on docs/guides/singlestore/configuration/podtemplating/index.md:
```diff
@@ -600,7 +600,7 @@ Containers:
       /scripts from init-scripts (rw)
       /var/lib/memsql from data (rw)
-      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-htm2z (ro)
+      /var/run/secrets/kubernetes.io/serviceaccount from kube-api-access-htm2z (rotate-auth)
```
Fix this accidental change.  On lot of pages.

Bonusree committed:
fix unwanted mistakes

Signed-off-by: Bonusree <bonusree@appscode.com>
